# 💙SCUBA'S FIRST DATA ANALYSIS💙

In [82]:
import pandas as pd
import numpy as np
import matplotlib as plt

# 📌 데이터 정보

#### 전국 평균 분양가격(2013년 9월부터 2015년 8월까지)
* 전국 공동주택의 `3.3제곱미터`당 평균분양가격 데이터를 제공

#### 주택도시보증공사_전국 평균 분양가격(2019년 12월)
* 전국 공동주택의 연도별, 월별, 전용면적별 `제곱미터`당 평균분양가격 데이터를 제공
* 지역별 평균값은 단순 산술평균값이 아닌 가중평균값임

# 📌 데이터 불러오기

In [20]:
!ls

jupyter basic.ipynb
practice1_inflearn.ipynb
상가업소정보_201912_01.csv
전국 평균 분양가격(2013년 9월부터 2015년 8월까지).csv
주택도시보증공사_전국 평균 분양가격(2019년 12월).csv


In [21]:
import os
for dirname, _, filenames in os.walk('./'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

./jupyter basic.ipynb
./practice1_inflearn.ipynb
./상가업소정보_201912_01.csv
./전국 평균 분양가격(2013년 9월부터 2015년 8월까지).csv
./주택도시보증공사_전국 평균 분양가격(2019년 12월).csv
./.ipynb_checkpoints/jupyter basic-checkpoint.ipynb
./.ipynb_checkpoints/practice1_inflearn-checkpoint.ipynb
./.git/config
./.git/HEAD
./.git/description
./.git/info/exclude
./.git/hooks/commit-msg.sample
./.git/hooks/pre-rebase.sample
./.git/hooks/pre-commit.sample
./.git/hooks/applypatch-msg.sample
./.git/hooks/fsmonitor-watchman.sample
./.git/hooks/pre-receive.sample
./.git/hooks/prepare-commit-msg.sample
./.git/hooks/post-update.sample
./.git/hooks/pre-merge-commit.sample
./.git/hooks/pre-applypatch.sample
./.git/hooks/pre-push.sample
./.git/hooks/update.sample


In [22]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings

warnings.filterwarnings('ignore')

In [23]:
df_first = pd.read_csv("./전국 평균 분양가격(2013년 9월부터 2015년 8월까지).csv", engine = "python", encoding="cp949")

In [24]:
df_first.head()

지역  2013년12월  2014년1월  2014년2월  2014년3월  2014년4월  2014년5월  2014년6월  \
0  서울     18189    17925    17925    18016    18098    19446    18867   
1  부산      8111     8111     9078     8965     9402     9501     9453   
2  대구      8080     8080     8077     8101     8267     8274     8360   
3  인천     10204    10204    10408    10408    10000     9844    10058   
4  광주      6098     7326     7611     7346     7346     7523     7659   

   2014년7월  2014년8월  ...  2014년11월  2014년12월  2015년1월  2015년2월  2015년3월  \
0    18742    19274  ...     20242     20269    20670    20670    19415   
1     9457     9411  ...      9208      9208     9204     9235     9279   
2     8360     8370  ...      8439      8253     8327     8416     8441   
3     9974     9973  ...     10020     10020    10017     9876     9876   
4     7612     7622  ...      7752      7748     7752     7756     7861   

   2015년4월  2015년5월  2015년6월  2015년7월  2015년8월  
0    18842    18367    18374    18152    18443  
1     9327     9345     9515     9559     9581  
2     8446     8568     8542     8542     8795  
3     9938    10551    10443    10443    10449  
4     7914     7877     7881     8089     8231  

[5 rows x 22 columns]

In [25]:
df_last = pd.read_csv("./주택도시보증공사_전국 평균 분양가격(2019년 12월).csv", engine = "python", encoding = "cp949")

In [26]:
df_last.head()

지역명               규모구분    연도   월 분양가격(㎡)
0  서울                 전체  2015  10    5841
1  서울         전용면적 60㎡이하  2015  10    5652
2  서울   전용면적 60㎡초과 85㎡이하  2015  10    5882
3  서울  전용면적 85㎡초과 102㎡이하  2015  10    5721
4  서울        전용면적 102㎡초과  2015  10    5879

In [27]:
df_last.tail()

지역명               규모구분    연도   월 분양가격(㎡)
4330  제주                 전체  2019  12    3882
4331  제주         전용면적 60㎡이하  2019  12     NaN
4332  제주   전용면적 60㎡초과 85㎡이하  2019  12    3898
4333  제주  전용면적 85㎡초과 102㎡이하  2019  12     NaN
4334  제주        전용면적 102㎡초과  2019  12    3601

# 📌 데이터 요약하기

In [29]:
df_last.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4335 entries, 0 to 4334
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   지역명      4335 non-null   object
 1   규모구분     4335 non-null   object
 2   연도       4335 non-null   int64 
 3   월        4335 non-null   int64 
 4   분양가격(㎡)  4058 non-null   object
dtypes: int64(2), object(3)
memory usage: 169.5+ KB


# 📌 결측치 보기

In [31]:
True ==1

True

In [32]:
False == 0

True

In [35]:
df_last.isnull()

지역명   규모구분     연도      월  분양가격(㎡)
0     False  False  False  False    False
1     False  False  False  False    False
2     False  False  False  False    False
3     False  False  False  False    False
4     False  False  False  False    False
...     ...    ...    ...    ...      ...
4330  False  False  False  False    False
4331  False  False  False  False     True
4332  False  False  False  False    False
4333  False  False  False  False     True
4334  False  False  False  False    False

[4335 rows x 5 columns]

In [34]:
df_last.isnull().sum()

지역명          0
규모구분         0
연도           0
월            0
분양가격(㎡)    277
dtype: int64

# 📌 데이터 전처리 과정

## 1) 데이터 타입 변경

In [37]:
df_last["분양가격(㎡)"]

0       5841
1       5652
2       5882
3       5721
4       5879
        ... 
4330    3882
4331     NaN
4332    3898
4333     NaN
4334    3601
Name: 분양가격(㎡), Length: 4335, dtype: object

### 🔑 분양가격의 값이 숫자인데, datatype이 object로 되어 있어서, 추후 계산의 용이성을 위해 숫자 데이터로 바꿔줄 필요가 있다. 

In [40]:
pd.to_numeric(df_last["분양가격(㎡)"], errors='coerce')

0       5841.0
1       5652.0
2       5882.0
3       5721.0
4       5879.0
         ...  
4330    3882.0
4331       NaN
4332    3898.0
4333       NaN
4334    3601.0
Name: 분양가격(㎡), Length: 4335, dtype: float64

In [42]:
type(pd.np.nan)

float

In [43]:
df_last["분양가격"]=pd.to_numeric(df_last["분양가격(㎡)"], errors='coerce')

In [47]:
df_last["분양가격"]

0       5841.0
1       5652.0
2       5882.0
3       5721.0
4       5879.0
         ...  
4330    3882.0
4331       NaN
4332    3898.0
4333       NaN
4334    3601.0
Name: 분양가격, Length: 4335, dtype: float64

In [49]:
df_last["분양가격"].mean()

3238.128632802628

## 2) 평당 분양가격 구하기

In [61]:
df_first.head(1)

지역  2013년12월  2014년1월  2014년2월  2014년3월  2014년4월  2014년5월  2014년6월  \
0  서울     18189    17925    17925    18016    18098    19446    18867   

   2014년7월  2014년8월  ...  2014년11월  2014년12월  2015년1월  2015년2월  2015년3월  \
0    18742    19274  ...     20242     20269    20670    20670    19415   

   2015년4월  2015년5월  2015년6월  2015년7월  2015년8월  
0    18842    18367    18374    18152    18443  

[1 rows x 22 columns]

In [62]:
df_last["평당분양가격"] = df_last["분양가격"] * 3.3

In [58]:
df_last.head(1)

지역명 규모구분    연도   월 분양가격(㎡)    분양가격   평당분양가격
0  서울   전체  2015  10    5841  5841.0  19275.3

## 3) 분양가격 요약하기

In [59]:
df_last.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4335 entries, 0 to 4334
Data columns (total 7 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   지역명      4335 non-null   object 
 1   규모구분     4335 non-null   object 
 2   연도       4335 non-null   int64  
 3   월        4335 non-null   int64  
 4   분양가격(㎡)  4058 non-null   object 
 5   분양가격     3957 non-null   float64
 6   평당분양가격   3957 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 237.2+ KB


In [67]:
df_last['분양가격(㎡)'].describe()

count     4058
unique    1753
top       2221
freq        17
Name: 분양가격(㎡), dtype: object

### 🔑 Count의 차이 
* 공백값(결측치)도 카운트하기 때문에, 위와 아래의 데이터 카운트값이 달라졌다.

### 🔑 데이터 타입 = object일때 다음과 같이 데이터의 요약을 이해한다.
* Uique -> string data에서 유니크한 데이터가 몇 개가 있는지 알 수 있다.
* top -> 가장 빈번하고 있는 데이터(문자)
* freqeuncy -> 가장 빈번하게 등장하는 데이터(문자)가 몇 번 등장하는지 

In [68]:
df_last['분양가격'].describe()

count     3957.000000
mean      3238.128633
std       1264.309933
min       1868.000000
25%       2441.000000
50%       2874.000000
75%       3561.000000
max      12728.000000
Name: 분양가격, dtype: float64

### 🔑 데이터 타입 = float64일때, 다음과 같이 데이터의 요약을 이해한다. 
* 64는 길이를 나타낸다.
* mean = 평균값
* std = 표준편차
* min = 최솟값
* 25% = 1사분위 수 
* 50% = 2사분위 수, 중앙값, Median값
* 75% = 3사분위 수
* max = 최댓값

`describe해서 주의해서 봐야할 것`
* mean (평균값)과 50% (중앙값, median)값이 얼마나 차이가 나는지를 확인한다. max를 보면, 최댓값 때문에 평균값이 더 높게 나타난다 라는 것을 위 데이터에서는 확인할 수 있다.   

    

## 4) 규모 구분을 전용면적 컬럼으로 변경

In [73]:
df_last

지역명               규모구분    연도   월 분양가격(㎡)    분양가격   평당분양가격
0     서울                 전체  2015  10    5841  5841.0  19275.3
1     서울         전용면적 60㎡이하  2015  10    5652  5652.0  18651.6
2     서울   전용면적 60㎡초과 85㎡이하  2015  10    5882  5882.0  19410.6
3     서울  전용면적 85㎡초과 102㎡이하  2015  10    5721  5721.0  18879.3
4     서울        전용면적 102㎡초과  2015  10    5879  5879.0  19400.7
...   ..                ...   ...  ..     ...     ...      ...
4330  제주                 전체  2019  12    3882  3882.0  12810.6
4331  제주         전용면적 60㎡이하  2019  12     NaN     NaN      NaN
4332  제주   전용면적 60㎡초과 85㎡이하  2019  12    3898  3898.0  12863.4
4333  제주  전용면적 85㎡초과 102㎡이하  2019  12     NaN     NaN      NaN
4334  제주        전용면적 102㎡초과  2019  12    3601  3601.0  11883.3

[4335 rows x 7 columns]

In [72]:
df_last["규모구분"].unique()

array(['전체', '전용면적 60㎡이하', '전용면적 60㎡초과 85㎡이하', '전용면적 85㎡초과 102㎡이하',
       '전용면적 102㎡초과'], dtype=object)

### 🔑 데이터 텍스트 변경
* str.replace()를 이용해 글자를 변경하고,.strip()을 이용해 앞뒤 여백을 제거한다.

In [95]:
df_last["전용면적"] = df_last["규모구분"].str.replace("전용면적", "")
df_last["전용면적"] = df_last["전용면적"].str.replace("초과","~")
df_last["전용면적"] = df_last["전용면적"].str.replace("이하","")
df_last["전용면적"] = df_last["전용면적"].str.replace(" ","").str.strip()

In [106]:
df_last["전용면적"]

0             전체
1            60㎡
2        60㎡~85㎡
3       85㎡~102㎡
4          102㎡~
          ...   
4330          전체
4331         60㎡
4332     60㎡~85㎡
4333    85㎡~102㎡
4334       102㎡~
Name: 전용면적, Length: 4335, dtype: object

In [107]:
df_last.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4335 entries, 0 to 4334
Data columns (total 8 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   지역명      4335 non-null   object 
 1   규모구분     4335 non-null   object 
 2   연도       4335 non-null   int64  
 3   월        4335 non-null   int64  
 4   분양가격(㎡)  4058 non-null   object 
 5   분양가격     3957 non-null   float64
 6   평당분양가격   3957 non-null   float64
 7   전용면적     4335 non-null   object 
dtypes: float64(2), int64(2), object(4)
memory usage: 271.1+ KB


In [110]:
df_last = df_last.drop(['규모구분', '분양가격(㎡)'], axis=1)

### 🔑 데이터 드롭하기 

* .drop을 이용해 행/열을 제거한다.
* axis = 0 : 행
* axis = 1 : 열

In [112]:
df_last 

지역명    연도   월    분양가격   평당분양가격      전용면적
0     서울  2015  10  5841.0  19275.3        전체
1     서울  2015  10  5652.0  18651.6       60㎡
2     서울  2015  10  5882.0  19410.6   60㎡~85㎡
3     서울  2015  10  5721.0  18879.3  85㎡~102㎡
4     서울  2015  10  5879.0  19400.7     102㎡~
...   ..   ...  ..     ...      ...       ...
4330  제주  2019  12  3882.0  12810.6        전체
4331  제주  2019  12     NaN      NaN       60㎡
4332  제주  2019  12  3898.0  12863.4   60㎡~85㎡
4333  제주  2019  12     NaN      NaN  85㎡~102㎡
4334  제주  2019  12  3601.0  11883.3     102㎡~

[4335 rows x 6 columns]

In [111]:
df_last.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4335 entries, 0 to 4334
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   지역명     4335 non-null   object 
 1   연도      4335 non-null   int64  
 2   월       4335 non-null   int64  
 3   분양가격    3957 non-null   float64
 4   평당분양가격  3957 non-null   float64
 5   전용면적    4335 non-null   object 
dtypes: float64(2), int64(2), object(2)
memory usage: 203.3+ KB


### 🔑 데이터 드롭 후 메모리 사용량 비교해보기

* 메모리 사용량이 271.1 + KB 에서 203.3 + KB로 줄어들었다. columns를 적당히 잘 통합/변경하여 메모리 사용량을 효과적으로 줄일 수 있다는 것을 알 수 있다. 

In [113]:
df_last.head()

지역명    연도   월    분양가격   평당분양가격      전용면적
0  서울  2015  10  5841.0  19275.3        전체
1  서울  2015  10  5652.0  18651.6       60㎡
2  서울  2015  10  5882.0  19410.6   60㎡~85㎡
3  서울  2015  10  5721.0  18879.3  85㎡~102㎡
4  서울  2015  10  5879.0  19400.7     102㎡~

# 📌 groupby 로 데이터 집계하기

### 🔑 1) 그룹화 할 colunm 중 index로 사용할 column, 계산할 column값, 실행할 연산 순으로 써준다.

#### `df.groupby(["index 컬럼값"])["계산할 컬럼값"].실행할연산()`

In [122]:
df_last.groupby(["지역명"])['평당분양가격'].mean()

지역명
강원     7890.750000
경기    13356.895200
경남     9268.778138
경북     8376.536515
광주     9951.535821
대구    11980.895455
대전    10253.333333
부산    12087.121200
서울    23599.976400
세종     9796.516456
울산    10014.902013
인천    11915.320732
전남     7565.316532
전북     7724.235484
제주    11241.276712
충남     8233.651883
충북     7634.655600
Name: 평당분양가격, dtype: float64

In [128]:
# 전용면적으로 분양가격의 평균을 구한다. 

df_last.groupby(["전용면적"])["분양가격"].mean()

전용면적
102㎡~       3490.213828
60㎡         3143.981037
60㎡~85㎡     3112.436385
85㎡~102㎡    3362.908962
전체          3113.965517
Name: 분양가격, dtype: float64

### 🔑 2) index 값을 2개 이상 넣고 싶을 때는 앞쪽 ([ ]) 안에 추가해준다. 또, unstack()을 이용해서, 끝에 있는 Index 값을 위에 column으로 빼줄 수 있다. 

#### `df.groupby(["index 컬럼값1, index 컬럼값2 "])["계산할 컬럼 값"].실행할연산().unstack()`

In [131]:
# 지역명, 전용면적으로 평당분양가격의 평균을 구한다. 

df_last.groupby(["전용면적","지역명"])["평당분양가격"].mean().unstack()

지역명                강원         경기            경남           경북            광주  \
전용면적                                                                        
102㎡~     8311.380000  14771.790  10358.363265  9157.302000  11041.532432   
60㎡       7567.098000  13251.744   8689.175000  7883.172000   9430.666667   
60㎡~85㎡   7485.588000  12523.566   8618.676000  8061.372000   9910.692000   
85㎡~102㎡  8749.557143  13677.774  10017.612000  8773.814634   9296.100000   
전체        7477.536000  12559.602   8658.672000  8078.532000   9903.630000   

지역명                 대구            대전         부산         서울            세종  \
전용면적                                                                       
102㎡~     13087.338000  14876.871429  13208.250  23446.038  10106.976000   
60㎡       11992.068000   9176.475000  11353.782  23212.794   9323.927027   
60㎡~85㎡   11778.690000   9711.372000  11864.820  22786.830   9775.458000   
85㎡~102㎡  11140.642857   9037.430769  12072.588  25943.874   9847.926000   
전체        11771.298000   9786.018000  11936.166  22610.346   9805.422000   

지역명                 울산            인천         전남          전북            제주  \
전용면적                                                                        
102㎡~      9974.448000  14362.030435  8168.4900  8193.57000  10522.787234   
60㎡        9202.106897  11241.318000  7210.1700  7609.93200  14022.221053   
60㎡~85㎡   10502.531707  11384.406000  7269.2400  7271.35200  10621.314000   
85㎡~102㎡   8861.007692  11527.560000  7908.8625  8275.78125  10709.082353   
전체        10492.712195  11257.026000  7283.5620  7292.60400  10784.994000   

지역명                충남        충북  
전용면적                             
102㎡~     8689.169388  8195.352  
60㎡       7911.156000  7103.118  
60㎡~85㎡   7818.954000  7264.488  
85㎡~102㎡  9120.045000  8391.306  
전체        7815.324000  7219.014

In [134]:

df_last.groupby(["전용면적","연도","지역명"])["평당분양가격"].mean().unstack().round()

지역명                 강원       경기       경남       경북       광주       대구       대전  \
전용면적     연도                                                                    
102㎡~    2015   7164.0  12379.0   9984.0   7658.0      NaN   8892.0      NaN   
         2016   7590.0  13136.0   8911.0   8075.0  11114.0  10550.0  14282.0   
         2017   7768.0  13651.0   9237.0   8956.0  10658.0  15087.0  14282.0   
         2018   8770.0  15271.0   9737.0   9442.0   8622.0  12846.0  16094.0   
         2019   9359.0  17535.0  13840.0  10514.0  14209.0  15082.0  15066.0   
60㎡      2015   7557.0  10448.0   7391.0   7251.0   7589.0   9239.0   8530.0   
         2016   7044.0  11744.0   7991.0   7213.0   8354.0  10930.0   8736.0   
         2017   7089.0  12381.0   8499.0   7900.0   8559.0  11852.0   9253.0   
         2018   7693.0  14397.0   9266.0   8264.0   9548.0  12141.0   8282.0   
         2019   8406.0  15114.0   9434.0   8315.0  11286.0  13722.0  11493.0   
60㎡~85㎡  2015   7258.0  10730.0   7652.0   7166.0   8109.0   9320.0   8012.0   
         2016   7093.0  11236.0   7849.0   7396.0   8919.0  10372.0   8432.0   
         2017   7115.0  11746.0   8072.0   7771.0   9452.0  11463.0   8921.0   
         2018   7668.0  13157.0   8940.0   8478.0   9856.0  12079.0  10180.0   
         2019   8093.0  14338.0   9810.0   8800.0  11827.0  13789.0  11671.0   
85㎡~102㎡ 2015   6801.0  11370.0   9684.0   8074.0      NaN   8411.0   8121.0   
         2016   6906.0  11089.0   9885.0   8722.0   9134.0   9248.0   8351.0   
         2017   7270.0  11897.0  10006.0   9052.0   9442.0  10316.0   9071.0   
         2018   9284.0  15281.0   9677.0   8784.0      NaN  11360.0  10121.0   
         2019  10673.0  16872.0  10584.0   8765.0      NaN  13963.0      NaN   
전체       2015   7160.0  10377.0   7586.0   7171.0   8052.0   9232.0   8098.0   
         2016   7011.0  11220.0   7848.0   7361.0   8899.0  10310.0   8502.0   
         2017   7127.0  11850.0   8120.0   7795.0   9464.0  11456.0   9045.0   
         2018   7681.0  13186.0   9019.0   8505.0   9856.0  12076.0  10180.0   
         2019   8142.0  14469.0   9871.0   8857.0  11823.0  13852.0  11778.0   

지역명                 부산       서울       세종       울산       인천      전남      전북  \
전용면적     연도                                                                  
102㎡~    2015  12076.0  20879.0   9012.0   9282.0  11246.0  7308.0  7495.0   
         2016  12610.0  21549.0   8827.0   9903.0  12732.0  7335.0  7650.0   
         2017  12633.0  22158.0   9121.0  10268.0  14782.0  7549.0  7580.0   
         2018  13182.0  23283.0  10475.0      NaN  14840.0  8205.0  8825.0   
         2019  14644.0  27329.0  12196.0      NaN  16067.0  9747.0  8842.0   
60㎡      2015   9771.0  20122.0   8520.0   8349.0  11158.0  6597.0  6799.0   
         2016   9916.0  21744.0   8878.0   8641.0  10430.0  6632.0  6430.0   
         2017  10849.0  21897.0   9156.0   9538.0  10784.0  7202.0  7663.0   
         2018  12385.0  23395.0  10111.0      NaN  11510.0  7403.0  8392.0   
         2019  12618.0  26478.0  10111.0  10685.0  12225.0  7757.0  8162.0   
60㎡~85㎡  2015   9975.0  19593.0   8799.0  10050.0  10293.0  6307.0  6710.0   
         2016  10214.0  20788.0   8922.0  10239.0  10479.0  6481.0  6440.0   
         2017  11651.0  20635.0   9058.0  11378.0  10744.0  7181.0  7040.0   
         2018  12926.0  22756.0  10376.0  10241.0  11631.0  7755.0  7559.0   
         2019  13123.0  27587.0  10930.0   9941.0  12903.0  7893.0  8168.0   
85㎡~102㎡ 2015   9758.0  21259.0   8744.0   9105.0  11698.0  7465.0  7844.0   
         2016  10548.0  24022.0   8801.0   8788.0  11322.0  7746.0  7596.0   
         2017  11093.0  23089.0   9194.0      NaN  11156.0  7745.0  7710.0   
         2018  12959.0  23688.0  10328.0      NaN  11139.0  8497.0  8471.0   
         2019  14187.0  33909.0  11290.0      NaN  12419.0  7712.0  9293.0   
전체       2015  10307.0  19725.0   8750.0  10053.0  10484.0  6317.0  6703.0   
         2016  10430.0  20663.0   8860.0  102

In [143]:
# 맨 끝에 .T(.Transpose)를 이용해 index 순서를 변경할 수도 있다. 
g = df_last.groupby(["연도","지역명"])["평당분양가격"].mean()
g.unstack().T

연도        2015          2016          2017          2018          2019
지역명                                                                   
강원    7188.060   7162.903846   7273.560000   8219.255000   8934.475000
경기   11060.940  11684.970000  12304.980000  14258.420000  15665.540000
경남    8459.220   8496.730000   8786.760000   9327.670000  10697.615789
경북    7464.160   7753.405000   8280.800000   8680.776923   9050.250000
광주    7916.700   9190.683333   9613.977551   9526.953333  12111.675000
대구    9018.900  10282.030000  12206.700000  12139.252632  14081.650000
대전    8190.600   8910.733333   9957.158491  10234.106667  12619.200000
부산   10377.400  10743.535000  11560.680000  12889.965000  13537.865000
서울   20315.680  21753.435000  21831.060000  23202.245000  28286.830000
세종    8765.020   8857.805000   9132.505556  10340.463158  11299.394118
울산    9367.600   9582.574138  10666.935714  10241.400000  10216.250000
인천   10976.020  11099.055000  11640.600000  11881.532143  13249.775000
전남    6798.880   6936.600000   7372.920000   7929.845000   8219.275862
전북    7110.400   6906.625000   7398.973585   8174.595000   8532.260000
제주    7951.075   9567.480000  12566.730000  11935.968000  11828.469231
충남    7689.880   7958.225000   8198.422222   8201.820000   8748.840000
충북    6828.800   7133.335000   7473.120000   8149.295000   7970.875000

# 📌 pivot table로 데이터 집계하기

### 🔑 어떤 점이 groupby 어떤 공통점/차이점이 있는가? 
* `groupby로 했던 작업을 pivot_table로 똑같이 작업할 수 있다.`
* `groupby는 .unstack을 설정해주지 않는 한, storage형태로 데이터가 출력된다면, pivot_table은 dataframe의 형태로 데이터가 출력된다는 차이점이 있다. `

In [142]:
pd.pivot_table(df_last, index=["지역명"], values=["평당분양가격"], aggfunc="sum")

평당분양가격
지역명           
강원   1909561.5
경기   3339223.8
경남   2289388.2
경북   2018745.3
광주   2000258.7
대구   2899376.7
대전   2030160.0
부산   3021780.3
서울   5899994.1
세종   2321774.4
울산   1492220.4
인천   2931168.9
전남   1876198.5
전북   1915610.4
제주   2461839.6
충남   1967842.8
충북   1908663.9